### PR TITLE
More prep work for advancing protocol versions

### PR DESCRIPTION
**Pull request by mystenmark**
This PR contains the bits of #8835 that are worth merging.

### CODE REVIEW

File: crates/sui-types/src/error.rs [hunk 1]
```diff
@@ -428,13 +427,7 @@ pub enum SuiError {
     SuiSystemStateUnexpectedVersion,
 
     #[error("Message version is not supported at the current protocol version")]
```
Comment (amnn) [inline crates/sui-types/src/error.rs:429]: nit: Can we expose the `error` message in the `thiserror` output?

```suggestion
    #[error("Message version is not supported at the current protocol version: {error}")]
```

File: crates/sui/tests/protocol_version_tests.rs [hunk 2]
```diff
@@ -305,7 +316,7 @@ mod sim_only_tests {
             }
         }));
 
-        monitor_version_change(&test_cluster, 2 /* expected proto version */).await;
+        monitor_version_change(&test_cluster, FINISH /* expected proto version */).await;
```
Comment (amnn) [inline crates/sui/tests/protocol_version_tests.rs:319]: Is it worth standardising on calls to `expect_upgrade_succeeded` and `expect_upgrade_failed`?

**Review by amnn — COMMENTED** [review]
Changes LGTM, but I'm not the expert for most of this code, so I'll defer to @lxfind for a final look, but let me know if I should stamp to unblock.

**Comment by lxfind** [comment]
Most changes seem to be non-functional? i.e. just set up boiler code for upgrades

**Review by lxfind — APPROVED** [review]

**Pull request merged**